### ISSUE TITLE
Pytlint under Python3 only displays the first error

### ISSUE THREAD

**Issue by DonJayamanne** [PyCQA/pylint#1021]
### Steps to reproduce
1. Create a python script with the following code:   
```python

print 1234
print "abcd"

x = y
print y
```
2. Run Pylint against the above file under Python 2.7 
3. Then run Pylint against the above file under Python 2.7 

### Current behavior in Python 2.7
```
No config file found, using default configuration
************* Module simulate
C:  1, 0: Missing module docstring (missing-docstring)
C:  5, 0: Invalid constant name "x" (invalid-name)
E:  5, 4: Undefined variable 'y' (undefined-variable)
E:  6, 6: Undefined variable 'y' (undefined-variable)
```

### Current behavior in Python 3.5
```
No config file found, using default configuration
************* Module simulate
E:  2, 0: Missing parentheses in call to 'print' (syntax-error)
```

### Expected behavior
I would have expected the linter to identify other errors (those identified under Python 2.7) as well, instead of stopping at the first error.

### pylint --version output
```
No config file found, using default configuration
pylint 1.6.2, 
astroid 1.4.7
Python 3.5.1 (v3.5.1:37a07cee5969, Dec  5 2015, 21:12:44) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]
```

### Is this a bug or by design?
Certainly others would have noticed this issue.
Or is this a design change under Python 3.5?

**Comment by moylop260** [comment]
IMHO a syntax error is a exit early.
Because its so hard detect code tokens when the compiler fails by syntax error.

Could you try fix the syntax error to see if the other errors are detected?

**Comment by DonJayamanne** [comment]
**"Exit early"**. Does that mean the linter will identify other linter messages before it exits.
Consider the following sample code:
```python
x = y

print 1234
print "abcd"
```

You'd expect the linter runs from top to bottom and would display the following errors before coming to the "print" statement:
```
1,0,convention,C0111:Missing module docstring
1,0,convention,C0103:Invalid constant name "x"
1,4,error,E0602:Undefined variable 'y'
```

It looks like the linter is ignoring all errors and only displaying the syntax error.

**Comment by The-Compiler** [comment]
Your code is valid Python 2 code, but invalid Python 3 code.

pylint works by constructing an [abstract syntax tree](https://en.wikipedia.org/wiki/Abstract_syntax_tree) and then running checks on it. If there is a syntax error, it can't construct that tree, and thus no checks can run.

**Comment by DonJayamanne** [comment]
Got it, thanks.
